packages/core/src/com/initializers/iframe.ts

```diff
@@ -48,33 +47,57 @@ async function useWindow(com: Communication, host: WindowHost, instanceId: strin
     await com.envReady(instanceId);
 }
 
+const iframeReloadHandlers = new WeakMap<HTMLIFrameElement, () => void>();
+
 async function useIframe(com: Communication, host: HTMLIFrameElement, instanceId: string, src: string): Promise<void> {
     const win = host.contentWindow;
     if (!win) {
         throw new Error('cannot spawn detached iframe.');
     }
 
-    await changeLocation(win, host, instanceId, src);
+    const locationChanged = await changeLocation(win, host, instanceId, src);
     com.registerEnv(instanceId, win);
-    await com.envReady(instanceId);
+    if (locationChanged) {
+        await com.envReady(instanceId);
+    }
 
-    const reload = () => com.reconnectHandlers(instanceId);
-    host.addEventListener('load', () => {
-        com.envReady(instanceId).then(reload).catch(reportError);
-    });
+    const handleIframeReload = async () => {
+        await com.envReady(instanceId);
+        com.reconnectHandlers(instanceId);
+    };
+    const existingReloadHandler = iframeReloadHandlers.get(host);
+    if (existingReloadHandler) {
+        host.removeEventListener('load', existingReloadHandler);
+    }
+    iframeReloadHandlers.set(host, handleIframeReload);
+    host.addEventListener('load', handleIframeReload);
```

Comment:
Maybe there's a more elegant way to do this, I wanted to prevent attaching multiple 'load' event listeners if this function was called multiple times on the same iframe.